### PR TITLE
[MPLUGIN-432] Improve site build

### PR DESCRIPTION
**Pull request by slawekjaranowski**
- rename duplicate packages - cause NPE in javadoc on clean project
- remove reports defined in parent pom
- aggregate reports should be not inherited

### CODE REVIEW

**Review by gnodet — APPROVED** [review]

File: maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/DescriptorGeneratorMojo.java [hunk 1]
```diff
@@ -195,7 +194,7 @@
      * This may be relative to the root of the generated Maven site.
      * It does not need to exist yet at the time when this goal is executed.
      * Must end with a slash.
-     * <b>In case this is set the javadoc reporting goal should be executed prior to {@link PluginReport}.</b>
+     * <b>In case this is set the javadoc reporting goal should be executed prior to Plugin Report.</b>
```
Comment (kwin) [inline maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/DescriptorGeneratorMojo.java:197]: I would still put a javadoc reference there (to the new goal). This is linked to the javadoc page in the goal page.

**Review by kwin — APPROVED** [review]
Looks good apart from the minor comment regarding the javadoc link

File: maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/DescriptorGeneratorMojo.java (hunk 1 above)
Comment (slawekjaranowski) [reply maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/DescriptorGeneratorMojo.java:197]: I will do it with #156

**Comment by jira-importer** [comment]
Resolve #733

**Pull request merged**